pkg/kubelet/volume/volume_manager.go

```diff
@@ -185,10 +187,17 @@ type volumeManager struct {
 	// desiredStateOfWorldPopulator runs an asynchronous periodic loop to
 	// populate the desiredStateOfWorld using the kubelet PodManager.
 	desiredStateOfWorldPopulator populator.DesiredStateOfWorldPopulator
+
+	// mountedVolulmeToClean is the list of volumes already mounted before
+	// kubelet starts. They need to be cleaned up before reconciler runs
+	mountedVolumeToClean []operationexecutor.MountedVolume
 }
 
 func (vm *volumeManager) Run(stopCh <-chan struct{}) {
 	defer runtime.HandleCrash()
+	glog.Infof("Clean up already mounted volumes before Kubelet starts")
+	vm.cleanUpMountedVolumes()
+
```

Comment:
We already have cleanup logic. Only when kubelet restart, it loses information and cause the problem you have seen. That is why we want to add this extra clean up code before kubelet starts the normal cleanup loop